masu/external/notifications/aws/aws_notification_handler.py

```diff
@@ -0,0 +1,174 @@
+            body            (String) String representation of the request body JSON
+            validation      (Boolean) Flag to skip message signature validation (optional)
+
+        """
+        self._headers = header
+        self._body = body
+        self._msg_type = self._get_message_type()
+        if validation:
+            self._validate_message()
+        self._confirm_subscription()
+
+    def _validate_message(self):
+        """
+        Validate incoming message.
+
+        Verify that the message payload matches the provided signature.
+
+        Args:
+            None
+
+        Returns:
+            TBD
+
+        """
+        # Validate signature of request body.  Throw exception if verification fails
+        pass
+
+    def get_billing_source(self):
+        """
+        Retrieve the name of the S3 bucket from the SNS notification.
+
+        Args:
+            None
+
+        Returns:
+            (String) Name of the billing source from the notification
+
+        Raises:
+            (NotificationInterfaceFilter): Is raised when the notification is not for
+                                           a top-level manifest file.
+
+        """
+        s3_bucket_name = None
+        if self._msg_type == 'Notification' and self._body:
+            body_dict = json.loads(self._body)
+            message_dict = json.loads(body_dict['Message'])
+
+            object_key = message_dict['Records'][0]['s3']['object']['key']
+            if object_key.endswith('Manifest.json'):
+                if re.findall(self.UUID_REGEX, object_key):
+                    msg = 'Ignoring non-toplevel manifest file: {}'.format(object_key)
+                    raise NotificationInterfaceFilter(msg)
+
+            if not object_key.endswith('Manifest.json'):
+                msg = 'Ignoring non-manifest file: {}'.format(object_key)
+                raise NotificationInterfaceFilter(msg)
+
+            # There must be a more reliable way of getting bucket name..
+            s3_bucket_name = message_dict['Records'][0]['s3']['bucket']['name']
+
+        return s3_bucket_name if s3_bucket_name else None
+
+    def _get_message_type(self):
+        """
+        Retrieve the type of SNS message.
+
+        Args:
+            None
+
+        Returns:
+            (String) Message String
+            example:
+                Notification             - Cost Usage Report S3 bucket events
+                SubscriptionConfirmation - When subscribing Masu to the SNS Topic as
+                                           an HTTP endpoint
+        Raises:
+            (AWSNotificationHandlerError): Is raised when the notification contains a
+                                           message type other than 'SubscriptionConfirmation'
+                                           or 'Notification'
+
+        """
+        msg_type = None
+        for header in self._headers:
+            if header[0] == AWS_SNS_HEADER_MSG_TYPE:
```

Comment:
This could be simplified down to 
`filter(lambda header: header[0] == AWS_SNS_HEADER_MSG_TYPE, self._headers)`

Reply:
I don't know if simplified is the right word here, but yeah that will put it on one line :)